main.js

```diff
@@ -1,117 +1,107 @@
 
-    if(!userPassword.value){
-        showError("form_password","The user password is empty")
-    } else if(userPassword.value.length > 15 || userPassword.value.length < 4){
-        showError("form_password","The user password should be between 4 - 15")
-    } else if(!userPassword.value.match(nameRegex)){
-        showError("form_password","The user password contains characters that are not allowed")
+    if(!formValues.password){
+        setValidationError("errorPassword","The password is empty");
+    } else if(formValues.password.length > 15 || formValues.password.length < 4){
+        setValidationError("errorPassword","The password should be between 4 - 15");
+    } else if(!formValues.password.match(nameRegex)){
+        setValidationError("errorPassword","The password contains characters that are not allowed");   
+    } else if(formValues.password !== userInfo.password){
+        setValidationError("errorPassword","Password is not correct");
     } else
         return true;
 
+    showError("password_error",validationErrors.errorPassword);
+
     return false;
 }
 
-//Checking for a matching login and password
-function checkData() { 
-    if(usernameInput.value === userInfo.username &&
-        userPassword.value === userInfo.password){
-        return true;
-    } else showError("form_username","Username or password is not correct");
+function setValidationError(fieldNameKey, error){
+    validationErrors[fieldNameKey] = error;
+}
 
-    return false;
+function showError(elementId, textError){
+    document.getElementById(elementId).textContent = textError;
 }
 
-function fieldsClear() {
-    usernameInput.value = "";
-    userPassword.value = "";
+function finalValidation(e){
+    e.preventDefault();
+    
+    if (isValidUsername() & isValidPassword()) {
+        userInfo.token = "xxx"
+        alert("correct");
+    }  
 }
 
-// Output exeption message
-function showError(elementId, textError)
-{    
-    removeError();
-    const error = document.getElementById(elementId).appendChild(document.createElement("p"));
-    error.classList.add("error");
-    error.textContent = textError;
+function setFormValue(e){
+    formValues[e.target.id] = e.target.value;
 }
 
-// Delete exeption message
-function removeError(){
-    if(document.getElementById("form_username").contains(document.querySelector(".error"))){
-        document.getElementById("form_username").removeChild(document.querySelector(".error"));
-    } else if(document.getElementById("form_password").contains(document.querySelector(".error"))){
-        document.getElementById("form_password").removeChild(document.querySelector(".error"));
-    }
+function addTouchedElements(e) {
+    touchedElements.push(e.target);
 }
 
-// Creating html registration form
 function createLoginForm() {
-    document.body.append(createFormContainer());
+    document.body.append(createFormContainer(finalValidation));
 
     const parentElement = document.getElementById("login");
```

Comment:
remove this line

Hint: 
`createFormContainer(finalValidation) === parentElement`

Reply:
Done